bob-backend/src/pearl/group.rs

```diff
@@ -63,39 +64,37 @@ impl Group {
         }
     }
 
-    pub async fn run(&self, pp: impl Hooks) -> AnyResult<()> {
+    async fn run_under_reinit_lock(&self, pp: impl Hooks) -> AnyResult<()> {
         debug!("{}: read holders from disk", self);
         let config = self.settings.config();
-        let holders = config
+        let new_holders = config
             .try_multiple_times_async(
                 || self.read_vdisk_directory(),
                 "can't create pearl holders",
                 self.settings.config().fail_retry_timeout(),
             )
             .await
             .with_context(|| "backend pearl group read vdisk directory failed")?;
-        debug!("{}: count holders: {}", self, holders.len());
-        if !holders
-            .iter()
-            .any(|holder| holder.gets_into_interval(get_current_timestamp()))
-        {
-            self.create_current_pearl();
-        }
+        debug!("{}: count holders: {}", self, new_holders.len());
         debug!("{}: save holders to group", self);
-        self.add_range(holders).await;
+        let mut holders = self.holders.write().await;
+        holders.extend(new_holders).await;   
```

Comment:
In fact, I don't see any normal scenario where we need to keep old items in the `holders` collection. It might be good to clear `holders` right before calling `extend`